vscode-dotnet-runtime-library/src/Utils/TypescriptUtilities.ts

```diff
@@ -84,12 +84,13 @@ export async function executeWithLock<A extends any[], R>(eventStream: IEventStr
         eventStream?.post(new DotnetLockReleasedEvent(`Lock about to be released, But we never touched it (pkilled vscode?)`, new Date().toISOString(), lockPath, lockPath));
         if (lockfile.checkSync(lockPath))
         {
+            eventStream?.post(new DotnetLockReleasedEvent(`Lock about to be released, and checkSync showed it.`, new Date().toISOString(), lockPath, lockPath));
             lockfile.unlockSync(lockPath);
         }
         else
         {
             eventStream?.post(new DotnetLockReleasedEvent(`Lock is not owned by us, delete it`, new Date().toISOString(), lockPath, lockPath));
-            fs.unlinkSync(`${lockPath}.lock`);
+            fs.rmdirSync(`${lockPath}.lock`, { recursive: true });

```

Comment:
Do you need to check if the lock is actually a dir, like fstat or something? Why delete recursively? do you expect the lock dir to have content, and if so, is there a chance any of the content could be a symlink?

Reply:
The lock should always be a dir, and the recursively is to make sure in case the logic changes in the future from the lock library that we delete anything in the dir. At the moment, I don't expect there to be any content in the directory. But, that's a good question 🤔